examples/pytorch_lightning/swav_queue.py

```diff
@@ -0,0 +1,155 @@
+# Note: The model and training settings do not follow the reference settings
+# from the paper. The settings are chosen such that the example can easily be
+# run on a small dataset with a single GPU.
+
+import torch
+from torch import nn
+import torchvision
+import pytorch_lightning as pl
+
+from lightly.data import LightlyDataset
+from lightly.data import SwaVCollateFunction
+from lightly.loss import SwaVLoss
+from lightly.loss.memory_bank import MemoryBankModule
+from lightly.models.modules import SwaVProjectionHead
+from lightly.models.modules import SwaVPrototypes
+
+
+class SwaV(pl.LightningModule):
+    def __init__(
+        self,
+        num_ftrs,
+        out_dim,
+        n_prototypes,
+        n_queues,
+        queue_length=0,
+        start_queue_at_epoch=0,
+        n_steps_frozen_prototypes=0,
+    ):
```

Comment:
We usually hardcode the different parameters in the examples, see `examples/pytorch_lightning/swav.py`